smart-contract-verifier/src/solidity/solc_cli.rs

```diff
@@ -0,0 +1,493 @@
+    pub struct OutputContract {
+        #[serde(deserialize_with = "serde_helpers::deserialize_abi_string")]
+        pub abi: Vec<serde_json::Value>,
+        pub bin: String,
+        #[serde(rename = "bin-runtime")]
+        pub bin_runtime: String,
+    }
+
+    impl OutputContract {
+        fn keys() -> Vec<String> {
+            vec![
+                "abi".to_string(),
+                "bin".to_string(),
+                "bin-runtime".to_string(),
+            ]
+        }
+    }
+
+    impl TryFrom<OutputContract> for Contract {
+        type Error = SolcError;
+
+        fn try_from(output_contract: OutputContract) -> Result<Self, Self::Error> {
+            let contract = serde_json::json!({
+                "abi": output_contract.abi,
+                "evm": {
+                    "bytecode": {
+                        "object": output_contract.bin,
+                    },
+                    "deployedBytecode": {
+                        "object": output_contract.bin_runtime,
+                    },
+                },
+            });
+            let contract: Contract = serde_json::from_value(contract)?;
+            Ok(contract)
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+    pub struct OutputJson {
+        pub contracts: HashMap<String, OutputContract>,
+    }
+
+    impl TryFrom<OutputJson> for CompilerOutput {
+        type Error = SolcError;
+
+        fn try_from(output_json: OutputJson) -> Result<Self, Self::Error> {
+            let contracts = output_json
+                .contracts
+                .into_iter()
+                .map(|(name, output)| output.try_into().map(|contract| (name, contract)))
+                .collect::<Result<HashMap<String, Contract>, _>>()?;
+            let contracts_raw = serde_json::to_value(&contracts)?;
+            let compiler_output = serde_json::json!(
+                {
+                    "contracts": {
+                        // TODO: give filename, if only 1 file was provided
+                        "": contracts_raw
+                    }
+                }
+            );
+            let compiler_output = serde_json::from_value(compiler_output)?;
+            Ok(compiler_output)
+        }
+    }
+}
+
+fn compiler_error(message: String) -> ethers_solc::artifacts::Error {
+    ethers_solc::artifacts::Error {
+        source_location: None,
+        secondary_source_locations: vec![],
+        r#type: "".to_string(),
+        component: "".to_string(),
+        severity: Severity::Error,
+        error_code: None,
+        message,
+        formatted_message: None,
+    }
+}
+
+pub fn compile_using_cli(
+    solc: &PathBuf,
+    input: &CompilerInput,
+) -> Result<CompilerOutput, SolcError> {
+    let output = {
+        let input_args = types::InputArgs::try_from(input)?;
+        let input_files = types::InputFiles::try_from(input)?;
+        Command::new(solc)
```

Comment:
I suggest to use `tokio::process::Command::new` and mark the function as `async`

Reply:
make sense